cmd/operator/internal/controller/kubearchiveconfig_controller.go

```diff
@@ -162,18 +165,26 @@ func (r *KubeArchiveConfigReconciler) reconcileRole(ctx context.Context, kaconfi
 	return role, nil
 }
 
-func (r *KubeArchiveConfigReconciler) desiredRole(kaconfig *kubearchivev1alpha1.KubeArchiveConfig) (*rbacv1.Role, error) {
+func (r *KubeArchiveConfigReconciler) desiredRole(ctx context.Context, kaconfig *kubearchivev1alpha1.KubeArchiveConfig) (*rbacv1.Role, error) {
+	log := log.FromContext(ctx)
 	var rules []rbacv1.PolicyRule
 	for _, resource := range kaconfig.Spec.Resources {
 		apiGroup := ""
-		slash := strings.Index(resource.APIVersion, "/")
-		if slash > -1 {
-			apiGroup = resource.APIVersion[:slash]
+		apiVersion := resource.APIVersion
+		data := strings.Split(resource.APIVersion, "/")
+		if len(data) > 1 {
+			apiGroup = data[0]
+			apiVersion = data[1]
+		}
+		gvr, err := r.Mapper.RESTMapping(schema.GroupKind{Group: apiGroup, Kind: resource.Kind}, apiVersion)
```

Comment:
I think that we should add a comment to make it clear that gvr resources are expected to be the plural of the resource based on the kubernetes standard to use the plurals in the API endpoints.

Also, I think it's also worth to mention that k8s roles' rules expect the `resources` field to be filled with the plural lowercased version of the resource (as oposed to the singular PascalCase of the `kind` field of the resource)

Reply:
Fixed.